eduNEXT/platform-plugin-superset
# feat: add superset xblock

### Description

This PR creates an XBlock to integrate Superset with the course content.


Studio Preview:
![image](https://github.com/eduNEXT/platform-plugin-superset/assets/33465240/e7f7f529-5c8a-4b8a-811c-498cb6382219)
LMS View:
![image](https://github.com/eduNEXT/platform-plugin-superset/assets/33465240/9302fdb7-31ed-42a8-a642-cd3cc3a748f9)
Edit View:
![image](https://github.com/eduNEXT/platform-plugin-superset/assets/33465240/1646cee7-1bdc-4d22-8f97-69e4b2cb55f1)

### Testing instructions

Assuming you already configured Aspects with a version later of `0.55.0`:

- Install this XBlock
- Go to the studio and add `"superset"` to your Advanced Module List.
- Add the component to your course.
- Follow the instructions to edit the component settings. Verify everything is working.
- By default, it integrates the Instructor dashboard (this is just for development purposes, it will be removed in a following PR). You can go to Superset, go to the Operator Dashboard (or create a new empty dashboard) http://superset.local.overhang.io:8088/superset/dashboard/operator-dashboard/, click the three dots menu, and select `Embed Dashboard`:
![image](https://github.com/eduNEXT/platform-plugin-superset/assets/33465240/8ef06c31-e298-44d5-b5c4-6c0305f93162)
- Leave the `allowed domains` field empty and enable it.
- Copy the dashboard ID and update the configuration of the XBlock


**Merge checklist:**
Check off if complete *or* not applicable:
